src/HDF5.jl

```diff
@@ -1395,7 +1395,13 @@ function read{S<:String}(obj::DatasetOrAttribute, ::Type{Array{S}})
             end
         end
     end
-    h5t_close(memtype_id)
+    try
+        dspace = dataspace(obj)
```

Comment:
I see, thanks for pointing this out. I'm not sure if we should worry about performance at this point.